db/model/model.js

```diff
@@ -63,19 +63,72 @@ exports.fetchAllArticles = () => {
       throw err;
     });
 };
-
-exports.fetchCommentsByArticleId = (article_id) => {
+exports.checkIdExists = (article_id, res, next) => {
   return db
     .query(
       `
-  SELECT * FROM comments WHERE article_id = $1 
-  ORDER BY comments.created_at DESC;`,
+  SELECT * FROM comments WHERE article_id = $1;`,
+      [article_id]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        res.status(404).send("id not found");
+      }
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+exports.fetchCommentsByArticleId = (article_id, next) => {
+  return db
+    .query(
+      `
+  SELECT * FROM comments WHERE article_id = $1
+    ORDER BY comments.created_at DESC;`,
       [article_id]
     )
     .then(({ rows }) => {
       return rows;
     })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.fetchPostedComment = (article_id, username, body) => {
+  return db
+    .query(
+      `INSERT INTO comments 
+        (body, article_id, author)
+        VALUES ($1, $2, $3)
+        RETURNING *`,
+      [body, article_id, username]
+    )
+    .then(({ rows }) => {
+      return rows[0].body;
+    })
     .catch((err) => {
       throw err;
     });
 };
+
+exports.verifyDetails = (article_id, username, res, next) => {
+  return db
```

Comment:
This is a great model function to check for the username and article_id in the database, and you are using it very well in the controller too!
Alternatively, this table has _article_id_ and _username_ as foreign key constraints, which means that you can get by with just a _POST_ query alone, and psql will throw an error if they cannot be found in the database, which would save you a bit of work / code